utils/data_vis_utils.py

```diff
@@ -0,0 +1,432 @@
+        .mark_bar()
+        .encode(
+            x=alt.X("profile:O", title="Profile", axis=alt.Axis(labelAngle=0)),
+            y=alt.Y(f"{y_col}:Q", title=f"{energy_type} consumption (kWh)"),
+            color=alt.value(colour),
+            tooltip=[
+                alt.Tooltip("profile:O", title="Profile"),
+                alt.Tooltip(f"{y_col}:Q", title=title, format=".0f"),
+            ],
+        )
+        .properties(width=600, height=400)
+    )
+    return chart
+
+
+def plot_contextual_info(data: pd.DataFrame, variable: str, title: str) -> alt.Chart:
+    """
+    Creates a bar chart comparing the proportion of households with a specific
+    characteristic across different profiles.
+    It also adds a horizontal dashed line indicating the average value in the population.
+
+    Args:
+        data (pd.DataFrame): contextual data
+        variable (str): conrextual variable to plot (e.g. "income") withtout "counts_" or "proportion_" prefix
+        title (str): title of the chart
+
+    Returns:
+        alt.Chart: An Altair chart object showing the proportion of households with the specified characteristic
+    """
+    # In the contextual info df, the profile is stored as a string
+    profiles_as_strings = [str(p) for p in configs.profile_numbers]
+
+    # select only contextual information about profiles by select numbers only
+    # (when there are low counts for a number of profiles, contextual information
+    # is provided as an aggregate and "profile" might have values such as "All other profiles")
+    profiles_specific_data = data[data["profile"].isin(profiles_as_strings)]
+
+    # creating barchart with a bar for each profile
+    chart = (
+        alt.Chart(
+            profiles_specific_data, title=alt.TitleParams(text=title, anchor="middle")
+        )
+        .mark_bar()
+        .encode(
+            x=alt.X("profile:N", axis=alt.Axis(labelAngle=0)),
+            y=alt.Y(f"proportion_{variable}:Q", title="Percentage of households (%)"),
+            color=alt.value(NESTA_COLOURS[0]),
+            tooltip=[
+                alt.Tooltip("profile", title="Profile"),
+                alt.Tooltip(
+                    f"proportion_{variable}:Q",
+                    title="Percentage of households (%)",
+                    format=".0f",
+                ),
+            ],
+        )
+        .properties(width=600, height=400)
+    )
+
+    # adding a horizontal dashed line for average value in population
+    avg_population = round(
+        data[f"counts_{variable}"].sum() / data["number_households"].sum() * 100, 2
+    )
+    chart = chart + alt.Chart(
+        pd.DataFrame({"avg_population": [avg_population]})
+    ).mark_rule(strokeDash=[1, 0], strokeWidth=3, color=NESTA_COLOURS[10]).encode(
+        y=alt.Y("avg_population:Q"),
+        tooltip=[
+            alt.Tooltip("avg_population:Q", title="Population average", format=".2f")
+        ],
+    )
+    return chart
```

Comment:
```suggestion
    return (
        chart.configure_axis(labelFont="Arial", titleFont="Arial")
        .configure_title(font="Arial")
        .configure_legend(labelFont="Arial", titleFont="Arial")
    )
```

All of the text in the Altair charts were showing in my browser in a Times New Roman-esque type font. For consistency with the rest of the app, these lines can be added for each chart function to set the plot text to also be a sans serif type.

Reply:
as per our chat, these should appear as Averta. So i'll check if that's the case when we deploy the app. If not, then I'll fix it in a separate branch.